hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -87,6 +89,8 @@ public void initialize(String inputName, Map<String, String> properties) {
       this.conf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, properties.get(CatalogProperties.WAREHOUSE_LOCATION));
     }
 
+    this.filterIcebergTable = Boolean.parseBoolean(properties.get(CatalogProperties.FILTER_ICEBERG_TABLE));
```

Comment:
It isn't clear from the property name what this is doing. Looks like you're trying to avoid calling `getTableObjectsByName` because that is slow. But it is needed to check whether a table is Iceberg or not.

I think it is a good idea to be able to list all tables and not just Iceberg tables, so instead of a "filter" setting that is unclear about what is filtered out, let's change it to be a setting to return all tables vs just Iceberg tables. How about `list-all=true` to list all tables and not just Iceberg tables?

Reply:
that's a good suggestion, I'll change the property name to hive.catalog.list-all-tables.